tests/foreman/longrun/test_oscap.py

```diff
@@ -540,6 +540,121 @@ def test_positive_oscap_run_via_ansible(
         assert result is not None
 
 
+@pytest.mark.libvirt_content_host
+@pytest.mark.tier4
+def test_positive_oscap_run_via_ansible_bz_1814988(
+    module_org, default_proxy, content_view, lifecycle_env
+):
+    """End-to-End Oscap run via ansible
+
+    :id: 375f8f08-9299-4d16-91f9-9426eeecb9c5
+
+    :parametrized: yes
+
+    :setup: scap content, scap policy, host group
+
+    :steps:
+
+        1. Create a valid scap content
+        2. Import Ansible role theforeman.foreman_scap_client
+        3. Import Ansible Variables needed for the role
+        4. Create a scap policy with anisble as deploy option
+        5. Associate the policy with a hostgroup
+        6. Provision a host using the hostgroup
+        7. Harden the host by remediating it with DISA STIG security policy
+        8. Configure REX and associate the Ansible role to created host
+        9. Play roles for the host
+
+    :expectedresults: REX job should be success and ARF report should be sent to satellite
+
+    :BZ: 1814988
+
+    :CaseImportance: Critical
+    """
+    hgrp_name = gen_string('alpha')
+    policy_name = gen_string('alpha')
+    # Creates host_group for rhel7
+    make_hostgroup(
+        {
+            'content-source-id': default_proxy,
+            'name': hgrp_name,
+            'organizations': module_org.name,
+        }
+    )
+    # Creates oscap_policy.
+    scap_id, scap_profile_id = fetch_scap_and_profile_id(
+        OSCAP_DEFAULT_CONTENT['rhel7_content'], OSCAP_PROFILE['dsrhel7']
+    )
+    Ansible.roles_import({'proxy-id': default_proxy})
+    Ansible.variables_import({'proxy-id': default_proxy})
+    role_id = Ansible.roles_list({'search': 'foreman_scap_client'})[0].get('id')
+    make_scap_policy(
+        {
+            'scap-content-id': scap_id,
+            'hostgroups': hgrp_name,
+            'deploy-by': 'ansible',
+            'name': policy_name,
+            'period': OSCAP_PERIOD['weekly'].lower(),
+            'scap-content-profile-id': scap_profile_id,
+            'weekday': OSCAP_WEEKDAY['friday'].lower(),
+            'organizations': module_org.name,
+        }
+    )
+    with VirtualMachine(distro=DISTRO_RHEL7) as vm:
```

Comment:
Since we have begun VM conversion, please use the fixtures available in broker.py for this test.

Reply:
@JacobCallahan updated the pr. 
we don't have support for Rhel6 so I skipped VM conversion for one test from this file.
Also I added missing `configure_puppet` method from vm.py to hosts.py which is required for tests present in this file.